pkg/operator/config.go

```diff
@@ -60,86 +49,21 @@ type Images struct {
 	ClusterAPIControllerBareMetal string `json:"clusterAPIControllerBareMetal"`
 }
 
-// InstallConfig contains the mao relevant config coming from the install config, i.e provider
-type InstallConfig struct {
-	InstallPlatform `json:"platform"`
-}
-
-// InstallPlatform is the configuration for the specific platform upon which to perform
-// the installation. Only one of the platform configuration should be set
-type InstallPlatform struct {
-	// AWS is the configuration used when running on AWS
-	AWS interface{} `json:"aws,omitempty"`
-
-	// Libvirt is the configuration used when running on libvirt
-	Libvirt interface{} `json:"libvirt,omitempty"`
-
-	// OpenStack is the configuration used when running on OpenStack
-	OpenStack interface{} `json:"openstack,omitempty"`
-
-	// Kubemark is the configuration used when running with Kubemark
-	Kubemark interface{} `json:"kubemark,omitempty"`
-
-	// BareMetal is the configuration used when running on managed Bare Metal
-	BareMetal interface{} `json:"baremetal,omitempty"`
-
-	// None is the configuration used when running on unmanaged UPI
-	None interface{} `json:"none,omitempty"`
-}
-
-func getInstallConfig(client kubernetes.Interface) (*InstallConfig, error) {
-	cm, err := client.CoreV1().ConfigMaps(ClusterConfigNamespace).Get(ClusterConfigName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("failed getting clusterconfig %s/%s: %v", ClusterConfigNamespace, ClusterConfigName, err)
-	}
-
-	return getInstallConfigFromClusterConfig(cm)
-}
-
-// getInstallConfigFromClusterConfig builds an install config from the cluster config.
-func getInstallConfigFromClusterConfig(clusterConfig *corev1.ConfigMap) (*InstallConfig, error) {
-	icYaml, ok := clusterConfig.Data[InstallConfigKey]
-	if !ok {
-		return nil, fmt.Errorf("missing %q in configmap", InstallConfigKey)
-	}
-	var ic InstallConfig
-	if err := yaml.Unmarshal([]byte(icYaml), &ic); err != nil {
-		return nil, fmt.Errorf("invalid InstallConfig: %v yaml: %s", err, icYaml)
-	}
-	return &ic, nil
-}
-
-func getProviderFromInstallConfig(installConfig *InstallConfig) (Provider, error) {
-	v := reflect.ValueOf(installConfig.InstallPlatform)
-	var nonNilFields int
-
-	for i := 0; i < v.NumField(); i++ {
-		if v.Field(i).Interface() != nil {
-			nonNilFields = nonNilFields + 1
-		}
-		if nonNilFields > 1 {
-			return "", fmt.Errorf("more than one platform provider given")
-		}
-	}
-
-	if installConfig.AWS != nil {
+func getProviderFromInfrastructure(infra *configv1.Infrastructure) (Provider, error) {
+	switch infra.Status.Platform {
```

Comment:
can't this function now just return `infra.Status.Platform` and we completely drop `type Provider string`?

Reply:
didn't want to change more than required. :)